src/path.jl

```diff
@@ -533,8 +533,13 @@ julia> created(p"src/FilePathsBase.jl")
 ```
 """
 created(fp::AbstractPath) = stat(fp).ctime
-Base.isdir(fp::AbstractPath) = isdir(mode(fp))
+
+Base.isdir(fp::AbstractPath) = isdir(directoriestype(fp), fp)
+Base.isdir(::DirectoriesExplicit, fp::AbstractPath) = isdir(mode(fp))
```

Comment:
So the assumption is that if I have:
```
directoriestype(::Type{MyPath}) = DirectoriesImplicit()
```

Then I also need to implement `Base.isdir(::DirectoriesImplicit, fp::MyPath)`? So the benefit is just when you throw the `MethodError`... or are we expecting folks to extend `DirectoriesType` with some shared logic that could be shared with multiple key-value path types?